services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/application/ApplicationControllerITest.kt

```diff
@@ -333,6 +358,40 @@ class ApplicationControllerITest(@Autowired override val mockMvc: MockMvc) : Con
         verify { applicationService.sendApplication(1234, username) }
     }
 
+    @Test
+    @WithMockUser(username)
+    fun `Application can be linked to a hanke`() {
+        val hankeTunnus = "HAI-1234"
+        every { applicationService.getApplicationById(1234, username) } returns
+            AlluDataFactory.createApplication(id = 1234, hankeTunnus = hankeTunnus)
+        every { hankeService.getHankeId(hankeTunnus) } returns 42
+        every { permissionService.hasPermission(42, any(), any()) } returns true
+
+        get("/hakemukset/1234")
+            .andExpect(status().isOk)
+            .andExpect(jsonPath("$.hankeTunnus").value(hankeTunnus))
+
+        verify { applicationService.getApplicationById(1234, username) }
+        verify { hankeService.getHankeId(hankeTunnus) }
+        verify { permissionService.hasPermission(42, any(), any()) }
+    }
+
+    @Test
+    @WithMockUser(username)
+    fun `Creating an application without hanke permissions fails`() {
```

Comment:
The other endpoints should be tested as well.

Reply:
I'm not sure I understand your response correctly, but I think there should be explicit tests for what happens when the PermissionService returns false.

Reply:
There are more tests for that now.